pkg/kubeadm/master/manifests.go

```diff
@@ -54,47 +55,77 @@ const (
 //E0817 17:53:22.242658       1 event.go:258] Could not construct reference to: '&api.Endpoints{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"kube-scheduler", GenerateName:"", Namespace:"kube-system", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]api.OwnerReference(nil), Finalizers:[]string(nil)}, Subsets:[]api.EndpointSubset(nil)}' due to: 'selfLink was empty, can't make reference'. Will not report event: 'Normal' '%v became leader' 'moby'
 
 func WriteStaticPodManifests(s *kubeadmapi.KubeadmConfig) error {
-	staticPodSpecs := map[string]api.Pod{
-		// TODO this needs a volume
-		etcd: componentPod(api.Container{
+	// Placeholder for kube-apiserver pod spec command
+	apiServerCommand := getComponentCommand(apiServer, s)
+
+	// Check if the user decided to use an external etcd cluster
+	if len(s.InitFlags.API.ExternalEtcdEndpoints) > 0 {
+		arg := fmt.Sprintf("--etcd-servers=%s", strings.Join(s.InitFlags.API.ExternalEtcdEndpoints, ","))
+		apiServerCommand = append(apiServerCommand, arg)
+	} else {
+		apiServerCommand = append(apiServerCommand, "--etcd-servers=http://127.0.0.1:2379")
+	}
+
+	// Is etcd secured?
+	if s.InitFlags.API.ExternalEtcdCAFile != "" {
+		etcdCAFileArg := fmt.Sprintf("--etcd-cafile=%s", s.InitFlags.API.ExternalEtcdCAFile)
+		apiServerCommand = append(apiServerCommand, etcdCAFileArg)
+	}
+	if s.InitFlags.API.ExternalEtcdCertFile != "" && s.InitFlags.API.ExternalEtcdKeyFile != "" {
+		etcdClientFileArg := fmt.Sprintf("--etcd-certfile=%s", s.InitFlags.API.ExternalEtcdCertFile)
+		etcdKeyFileArg := fmt.Sprintf("--etcd-keyfile=%s", s.InitFlags.API.ExternalEtcdKeyFile)
+		apiServerCommand = append(apiServerCommand, etcdClientFileArg, etcdKeyFileArg)
+	}
+
+	// Prepare static pod specs
+	staticPodSpecs := make(map[string]api.Pod)
```

Comment:
Could you keep the standard components in-line here and add set `etcd` key down below? It'd make more sense to have it last, as it's conditional, plus it'd save a few lines :)